OSDOCS-8615: Configuring PID limits (#69352)
openshift/openshift-docs: modules/understanding-process-id-limits.adoc

// Module included in the following assemblies:
//
// * rosa_cluster_admin/rosa-configuring-pid-limits.adoc

:_mod-docs-content-type: CONCEPT
[id="understanding-process-id-limits_{context}"]
= Understanding process ID limits

> A process identifier (PID) is a unique identifier assigned by the Linux kernel to each process or thread currently running on a system. The number of processes that can run simultaneously on a system is limited to 4,194,304 by the Linux kernel, and may additionally be affected by limited access to other system resources such as memory, CPU, and disk space.

Reviewer: ```suggestion
A process identifier (PID) is a unique identifier assigned by the Linux kernel to each process or thread currently running on a system. The number of processes that can run simultaneously on a system is limited to 4,194,304 by the Linux kernel. This number might also be affected by limited access to other system resources such as memory, CPU, and disk space.
```
- [Per the ISG](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#m), only use "may" to indicate permission.
- For minimalism and clarity, break the long 2nd sentence into two sentences. 

cc @shipsing

> In {product-title}, there are two supported limits for PID usage to consider when you schedule work on your cluster:

Reviewer: ```suggestion
In {product-title}, consider these two supported limits for PID usage before you schedule work on your cluster:
```
Per [the ISG](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#t), avoid "there are" at the beginning of an independent clause.

cc @shipsing

> Maximum number of PIDs per pod (default: 4,096 in {product-title} 4.11 and higher)::

Reviewer 2: ```suggestion
Maximum number of PIDs per pod (default: 4,096 in {product-title} 4.11 and later)::
```

Reviewer: ```suggestion
* Maximum number of PIDs per pod. The default value is 4,096 in {product-title} 4.11 and later)
```
A few other changes in addition to the change by @shipsing that follows [the ISG guidance about referring to product versions](https://www.ibm.com/docs/en/ibm-style?topic=references-product-versions#references-to-earlier-or-later-versions).

> This is controlled by the `podPidsLimit` parameter set on the node.

Reviewer: ```suggestion
This limit is controlled by the `podPidsLimit` parameter set on the node.
```
- Per the ISG, avoid ambigous pronoun references. It's unclear here what "this" refers to. This what?

cc @shipsing

Maximum number of PIDs per node (default depends on link:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html-single/nodes/index#nodes-nodes-resources-configuring[node resources])::

> In OpenShift, this is controlled by the link:https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#system-reserved[`--system-reserved`] parameter, which reserves PIDs on each node based on the total resources of the node.

Reviewer 2: @bburt-rh, Is the formatting okay? I think the writer wants to highlight these points using a description list.

Reviewer: @shipsing @bhardesty - The two items here should be formatted as a bulleted list.

Reviewer: ```suggestion
Maximum number of PIDs per pod (default: 4,096 in {product-title} 4.11 and higher)::
This is controlled by the `podPidsLimit` parameter set on the node.

* Maximum number of PIDs per node. The default value depends on link:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html-single/nodes/index#nodes-nodes-resources-configuring[node resources]).
In {product-title}, this value is controlled by the link:https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#system-reserved[`--system-reserved`] parameter, which reserves PIDs on each node based on the total resources of the node.
```
- Format as a bullet point.
- Replace "OpenShift" with the correct attribute.
- Rewrite to avoid ambiguous pronoun reference.
- Also, as a point for future reference: Generally in the non-ROSA OCP docs, we don't cross-reference other OCP docs by using direct links the published HTML; we should always use xrefs to adoc files, though these xrefs can't be used in module files. However, for the ROSA docs, there's an exception because we can't link between different distros, such as from ROSA to OpenShift Container Platform. Thus, these can be external links and not cross-references. These external links use `link` instead of `xref` and need to be in distro-specific files or conditionalized to apply to the relevant distros.

Author: FYI, re: the link to "node resources" - I changed the module file name to start with the `sd-` prefix. The reason for the link is because it points to a topic that is not included in the ROSA docs. Because we don't have a good way to handle these scenarios, we typically just link to the OCP docs on the customer portal. I renamed the file to indicate that this is a Managed OpenShift (i.e. "service delivery") module, and is not intended for OCP.

> When a pod exceeds the allowed maximum number of PIDs per pod, the pod may stop functioning correctly and may be evicted from the node. See the Kubernetes documentation for more information on eviction: link:https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#eviction-signals-and-thresholds[Eviction signals and thresholds].

Reviewer: ```suggestion
When a pod exceeds the allowed maximum number of PIDs per pod, the pod might stop functioning correctly and might be evicted from the node. See link:https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#eviction-signals-and-thresholds[the Kubernetes documentation for eviction signals and thresholds] for more information.
```
- Don't use "may" to mean "might". Use "may" only to indicate permission
- Revise the "for more information" sentence for minimalism. My suggestion is one possibility.

> When a node exceeds the allowed maximum number of PIDs per node, the node can become unstable because new processes cannot have PIDs assigned. If existing processes cannot complete without creating additional processes, the entire node may become unusable and require reboot. This may result in data loss, depending on the processes and applications being run. Customer administrators and Red Hat Site Reliability Engineering are notified when this threshold is reached, and you will notice a "Worker node is experiencing PIDPressure" warning in cluster logs.

Reviewer 2: ```suggestion
When a node exceeds the allowed maximum number of PIDs per node, the node can become unstable because new processes cannot have PIDs assigned. If existing processes cannot complete without creating additional processes, the entire node may become unusable and require reboot. This may result in data loss, depending on the processes and applications being run. Customer administrators and Red Hat Site Reliability Engineering are notified when this threshold is reached, and you will notice a `Worker node is experiencing PIDPressure` warning in cluster logs.
```
We need to enclose a warning in backticks: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#system-messages

> When a node exceeds the allowed maximum number of PIDs per node, the node can become unstable because new processes cannot have PIDs assigned. If existing processes cannot complete without creating additional processes, the entire node may become unusable and require reboot. This may result in data loss, depending on the processes and applications being run. Customer administrators and Red Hat Site Reliability Engineering are notified when this threshold is reached, and you will notice a "Worker node is experiencing PIDPressure" warning in cluster logs.

Reviewer: ```suggestion
When a node exceeds the allowed maximum number of PIDs per node, the node can become unstable because new processes cannot have PIDs assigned. If existing processes cannot complete without creating additional processes, the entire node can become unusable and require reboot. This situation can result in data loss, depending on the processes and applications being run. Customer administrators and Red Hat Site Reliability Engineering are notified when this threshold is reached, and a `Worker node is experiencing PIDPressure` warning will appear in the cluster logs.
```
- Don't use "may" except to indicate permission.
- Don't use "this" without a clear reference.
